src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -110,4 +130,24 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
 		return bookRepository.save(book);
 	}
 
+	/**
+	 * This method is to find a book by isbn
+	 *
+	 * @param isbn: author of the book (Object)
+	 *
+	 * @return {@link BookInfoDto}
+	 */
+
+	@GetMapping("/isbn/{isbn}")
+	public ResponseEntity<Book> findBookByIsbn(@PathVariable String isbn) {
+
+		Optional<Book> book = bookRepository.findByIsbn(isbn);
+		if (book.isPresent()) {
```

Comment:
You can improve the readability of this code:

```
bookRepository.findByIsbn(isbn)
.map(book -> new ResponseEntity<>(book,HttpStatus.OK))
.orElseGet(new ResponseEntity<>(openLibraryDelegate.findBookByIsbn(isbn).get(),HttpStatus.CREATED));
```

Try it and let us know how it worked in both cases (when the book is found in the repository and when it is not found and has to retrieve it to the external API)

Also, is there any need for openLibraryDelegate.findBookByIsbn to return an Optional? I don't think you are using anything related to Optional in that method, you can return directly the Book.